src/variableDiagnosticsProvider.ts

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+import { workspace, languages, Diagnostic, DiagnosticSeverity, DiagnosticCollection, TextDocument, Range, Position, Disposable } from 'vscode';
+
+import { event } from "./events/requestVariableEvent";
+import { VariableProcessor } from "./variableProcessor";
+
+export class VariableDiagnosticsProvider {
+
+    private diagnosticCollection: DiagnosticCollection;
+
+    public activate(subscriptions: Disposable[]) {
```

Comment:
Is it possible to move the language diagnostic collection create and register text document related listeners in `extension.ts`. Since all of our language providers registered there. It will be easier to manage for organizing in one single place.

Reply:
Sure I'll look into it. This was just the recommended pattern when I looked at tutorials from Microsoft.